fearless/Common/Substrate/Types/CallCodingPath.swift

```diff
@@ -1,52 +1,75 @@
 import Foundation
 
-struct CallCodingPath: Equatable, Codable {
-    let moduleName: String
-    let callName: String
-}
+enum CallCodingPath: Equatable, Codable, CaseIterable {
+    #warning("Do not forget added new case in allCases")
+    static var allCases: [CallCodingPath] {
+        [
+            .transfer,
+            .transferKeepAlive,
+            .addMemo,
+            .nominationPoolJoin,
+            .createNominationPool,
+            .setPoolMetadata,
+            .poolBondMore,
+            .poolUnbond,
+            .claimPendingRewards,
+            .poolWithdrawUnbonded
+        ]
+    }
 
-extension CallCodingPath {
     var isTransfer: Bool {
         [.transfer, .transferKeepAlive].contains(self)
     }
 
-    static var transfer: CallCodingPath {
-        CallCodingPath(moduleName: "Balances", callName: "transfer")
-    }
-
-    static var transferKeepAlive: CallCodingPath {
-        CallCodingPath(moduleName: "Balances", callName: "transfer_keep_alive")
-    }
-
-    static var addMemo: CallCodingPath {
-        CallCodingPath(moduleName: "Crowdloan", callName: "add_memo")
+    var moduleName: String {
+        path.moduleName
     }
 
-    static var nominationPoolJoin: CallCodingPath {
-        CallCodingPath(moduleName: "NominationPools", callName: "Join")
+    var callName: String {
+        path.callName
     }
 
-    static var createNominationPool: CallCodingPath {
-        CallCodingPath(moduleName: "NominationPools", callName: "create")
+    var path: (moduleName: String, callName: String) {
+        switch self {
+        case .transfer:
+            return (moduleName: "Balances", callName: "transfer")
+        case .transferKeepAlive:
+            return (moduleName: "Balances", callName: "transfer_keep_alive")
+        case .addMemo:
+            return (moduleName: "Crowdloan", callName: "add_memo")
+        case .nominationPoolJoin:
+            return (moduleName: "NominationPools", callName: "Join")
+        case .createNominationPool:
+            return (moduleName: "NominationPools", callName: "Create")
+        case .setPoolMetadata:
+            return (moduleName: "NominationPools", callName: "set_metadata")
+        case .poolBondMore:
+            return (moduleName: "NominationPools", callName: "bond_extra")
+        case .poolUnbond:
+            return (moduleName: "NominationPools", callName: "unbond")
+        case .claimPendingRewards:
+            return (moduleName: "NominationPools", callName: "claim_payout")
+        case .poolWithdrawUnbonded:
+            return (moduleName: "NominationPools", callName: "withdraw_unbonded")
+        case let .fromInit(moduleName, callName):
```

Comment:
as I understand we need this case because there are places in our code that not cover by known enum cases (like .transfer or .addMemo). so I suggest to find those cases and cover with enum cases. it's important because that way we can avoid manual allCases usage.